smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -273,6 +273,21 @@ private void generateMux(GenerationContext context) {
        );
     }
 
+    private void generateOperationMux(GenerationContext context, OperationShape operation) {
+        TypeScriptWriter writer = context.getWriter();
+
+        writer.addImport("httpbinding", null, "@aws-smithy/server-common");
+
+        writer.openBlock("const mux = new httpbinding.HttpBindingMux<$S, $S>([", "]);",
+                context.getService().getId().getName(),
+                operation.getId().getName(),
+                () -> {
+                    HttpTrait httpTrait = operation.expectTrait(HttpTrait.class);
+                    generateUriSpec(context, operation, httpTrait);
```

Comment:
maybe this is overkill, but could this be shared so it's only defined once?

Reply:
Do you mean the UriSpec is only defined in TS once? These end up in different files, so I'd have to have a uri specs file, and generate and export a unique const per spec. Or generate a UriSpec factory that can take in a service and operation and export that. Does not seem worth the upside to me at th emoment.

Reply:
Yeah I meant define it once in the operation's file and then import it in the service. Doesn't really matter since it's pointless outside of the context of the mux anyway